### DISCUSSION TITLE
Cesium Tileset not working with Tile3DLayer

### DISCUSSION THREAD

**Discussion by anuraags** [visgl/deck.gl#4222] [Q&A]
Hi, I have a Cesium tileset (gltf 2.0 format) of the Sydney Opera house that I am trying to render using the Tile3DLayer component in deck.gl. 

The tileset is located here: https://drive.google.com/file/d/1py_O_5kJUePSEZs-FdiW-5EX3aBYxGqj/view?usp=sharing

If I drop the tiles into a local folder, set up a html file with a simple Cesium viewer and run it using python's SimpleHTTPServer everything works as expected. The opera house gets rendered at the correct location.

However if I attempt to use the Deck.GL Tile3DLayer nothing gets rendered. Interestingly if I go to the appropriate location it looks like the tileset hierarchy is properly getting retrieved but I can't see the model in the WebGL canvas.

Is there something wrong with the tileset? Can someone get this to work and share a snippet?

**Comment by Pessimistress** [comment]
Are you using deck.gl v8? There are known bugs with the 7.3 Tile3DLayer.

**Comment by anuraags** [comment]
I'm using the master version of the Tile3DLayer. I did a comparison between that one and version 8.0.12 and could not find any major differences.

**Comment by xintongxia** [comment]
I'll investigate this issue.

**Comment by anuraags** [comment]
Thank you! Please let me know if you need any code from me. I can post the cesium viewer code if necessary.

**Comment by xintongxia** [comment]
@georgios-uber I checked this dataset, it loaded the data and created `ScenegraphLayer`s, however the layers are not being rendered. Could you take a look?

**Comment by xintongxia** [comment]
@georgios-uber 

checkout the debug branch, `xx/scenegraph-debug`

Steps:

- bootstrap

```bash
# deck.gl root
yarn bootstrap
```

- start the app

```
cd test/apps/tiles-3d-scenegraph
yarn start-local
```

**Comment by anuraags** [comment]
@xintongxia @Pessimistress @georgios-uber You guys are awesome!

I tried the latest version "@loaders.gl/3d-tiles" (v2.0.4) and the model is now loading (which is great!) but the textures are not being rendered. If I set the lighting to "flat" for the scene graph layer the model is coming up but it's all white.

If I go to the chrome console I can see the textures coming through over the network but it looks like they aren't being applied to the model?

**Comment by georgioskarnas** [comment]
> If I go to the chrome console I can see the textures coming through over the network but it looks like they aren't being applied to the model?

The problem is we do not implement `KHR_techniques_webgl`. I'm working to get it implemented.

**Comment by anuraags** [comment]
Thanks @georgios-uber! 

Just to give a bit of background, I work at Nearmap. I'm trying to push the company to upgrade our tilesets to the latest specs and I wanted to use deck.gl to demonstrate the many benefits of upgrading.

We are also currently using vanilla cesium tilesets which don't confer the vast amount of benefits uber's vis.gl libraries do.

By the way I tried using gltf-pipeline to convert the mesh to draco format but large parts of the mesh were missing. I'll have a talk to cesium as well and see if they can help out. 

Thank you for helping me out!

**Comment by ibgreen** [comment]
@anuraags great to see nearmap working on getting your tilesets updated. I personally think nearmap would be better off avoiding optional glTF extensions like KHR_techniques_webgl and generate tilesets that render with standard glTF 2.0. That will increase the compatibility of your tilesets. If not, it would be interesting to hear why nearmap feels you need custom shaders rather than just using PBR materials.

While we recommend avoiding extensions in general, we do strongly support the glTF draco extension so if you have draco encoded tilesets that don’t load correctly in deck.gl, please share a tile so that we can debug.
